### PR TITLE
Envoy Custom Routing via Dynamic Forward Proxy & Lua

### PR DESCRIPTION
**Pull request by foxish**
Example for using custom routing with envoy with Signadot

### CODE REVIEW

File: experimental/custom-routing-with-envoy/README.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+This sets up [httpbin](https://httpbin.org/) in the cluster in the namespace
+httpbin. The httpbin application listens on port 80 and has an envoy sidecar
+that is running on port 8080 which is used for all communications in the
+cluster. 
+
+Once you have a signadot organization, connected a cluster and configured your
+signadot CLI, you can create a sandbox as shown below by running:
+
+```
+signadot sandbox apply -f ./sandbox-baseline.yaml
```
Comment (daniel-de-vera) [inline experimental/custom-routing-with-envoy/README.md:23]: I would make the `cluster` a templated variable, and tell the user that he has to replace it with the corresponding value:

`signadot sandbox apply -f ./sandbox-baseline.yaml --set cluster=<your-cluster-name>`

File: experimental/custom-routing-with-envoy/README.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+![Baseline](baseline.png)
+
+Note how it's running v0.9.0 of the application. This is the baseline version of
+the application.
+
+Now, we can create the second sandbox that is running a forked version of the
+httpbin application by running the following:
+
+```
+signadot sandbox apply -f ./sandbox.yaml
```
Comment (daniel-de-vera) [inline experimental/custom-routing-with-envoy/README.md:51]: Same as above.

**Review by daniel-de-vera — APPROVED** [review]
Overall looks good to me (except for the details I commented above, it works fine).

I wonder if it would be useful to provide a way (maybe via sandbox spec) to not fork certain containers. Because in this case, there is no need for having the sidecar in the forks (similarly to what we do with our sidecar, which is only present in the baseline deployments).

File: experimental/custom-routing-with-envoy/README.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+This example explains how you can use Signadot for realizing custom routing using envoy.
+
+## Set up
+
+To set up the demo application, you can run the following in your cluster:
+
+```
+kubectl create ns httpbin
+kubectl apply -f k8s/
```
[the remark this answers is not in the document]
Comment (foxish) [reply experimental/custom-routing-with-envoy/README.md:11]: Good catch, thanks! Fixed

**Comment by foxish** [comment]
> I wonder if it would be useful to provide a way (maybe via sandbox spec) to not fork certain containers. Because in this case, there is no need for having the sidecar in the forks (similarly to what we do with our sidecar, which is only present in the baseline deployments).

I think that would be useful. However, in a practical scenario, I think it's common to have sidecars injected via an admission webhook, which means it won't be in the PodTemplate at all, and therefore not automatically in the forked pod (unless the admission controller chooses to inject it in again).

**Pull request merged**